src/azure-cli/azure/cli/command_modules/storage/_validators.py

```diff
@@ -1235,6 +1235,28 @@ def as_user_validator(namespace):
                 None, "incorrect usage: specify '--auth-mode login' when as-user is enabled")
 
 
+def validator_change_feed_retention_days(namespace):
+    enable = namespace.enable_change_feed
+    days = namespace.change_feed_retention_days
+
+    from azure.cli.core.azclierror import InvalidArgumentValueError
+    if enable is False and days is not None:
+        raise InvalidArgumentValueError("incorrect usage: "
+                                        "'--change-feed-retention-days' is invalid "
+                                        "when '--enable-change-feed' is set to false")
+    if enable is None and days is not None:
+        raise InvalidArgumentValueError("incorrect usage: "
+                                        "please specify '--enable-change-feed true' if you "
+                                        "want to set the value for '--change-feed-retention-days'")
```

Comment:
If my storage account change feed is already enabled but I only want to change days for exiting policy, it will be blocked by this check, right?

Reply:
I keep align with DeleteRetentionPolicy that if `retention_days` given, `enable` should be explicitly `True`, see Line1253-Line1256:
https://github.com/Azure/azure-cli/blob/20963c0c6dc1c2e89ccfe3f707311b8901d15cbf/src/azure-cli/azure/cli/command_modules/storage/_validators.py#L1238-L1264

Reply:
ok. Then make sense because they are in the same command.